python/ccf/read_ledger.py

```diff
@@ -14,18 +14,49 @@ def indent(n):
     return " " * n
 
 
-def stringify_bytes(bs):
-    s = bs.decode()
-    if s.isprintable():
-        return s
-    if len(bs) > 0 and len(bs) <= 8:
-        n = int.from_bytes(bs, byteorder="little")
-        return f"<u{8 * len(bs)}: {n}>"
-    return bs
+def fmt_unsigned_int_little(data):
+    return (
+        f'<u{8 * len(data)}: {int.from_bytes(data, byteorder="little", signed=False)}>'
+    )
+
+
+def fmt_hex(data):
+    return data.hex()
+
+
+def fmt_str(data):
+    return data.decode()
+
+
+def fmt_json(data):
+    return json.dumps(json.loads(data), indent=2)
+
 
+# Dictionary of store table name regex to format function
+default_format_rule = {"key": fmt_hex, "value": fmt_hex}
+default_tables_format_rules = {
+    "^public:ccf\.internal\..*$": {
+        "key": fmt_unsigned_int_little,
+        "value": fmt_json,
+    },
+    "^public:ccf\.gov\.(service|network|constitution).*$": {
+        "key": fmt_unsigned_int_little,
+        "value": fmt_json,
+    },
+    "^public:ccf\.gov\..*$": {"key": fmt_str, "value": fmt_json},
+    ".*": {"key": fmt_hex, "value": fmt_hex},
+}
```

Comment:
I think this should be a list of pairs, rather than a dict. We don't need random lookup, we just iterate, and that makes it clearer that the order is intentional and meaningful.

Reply:
Yes, you're right, I'll change that (`dict`s are ordered since Python3.7 btw).